### DISCUSSION TITLE
SUPnP prototype implementation

### DISCUSSION THREAD

**Discussion by romansko** [pupnp/pupnp#452] [Show and tell]
I implemented a prototype of the SUPnP protocol, which is presented in the paper "SUPnP: Secure Access and Service Registration for UPnP-Enabled Internet of Things" by Kayas, G., Hossain, M., Payton, J., and Islam, S. R.

I built the SUPnP protocol on top of `libupnp`. I added the `ENABLE_SUPNP` guard to compile the secure layer; without it, `libupnp` will be compiled normally. The guard can be enabled with `-DENABLE_SUPNP=ON` for CMake or `--enable-supnp` for the configure script.

Feel free to read more at my fork: https://github.com/romansko/supnp.

There is a Python simulation component for generating the artifacts required by the SUPnP protocol. For more info: https://github.com/romansko/supnp/tree/supnp/simulation.

**Comment by Vollstrecker** [comment]
Just on a first look, it seems that there a re many unneeded file for your env. I'm not sure if all the reformatting ends up in a pull request, but I'm sure it shouldn't, and our options all start with UPNP_ and are announced via option to be shown in a gui that's used for configuration.

**Comment by mrjimenez** [comment]
Awsome work. It would be nice to integrate that.
